contracts/withdrawer/MultichainWithdrawer.sol

```diff
@@ -253,22 +248,18 @@ contract MultichainWithdrawer is BoringOwnable {
             if (feesEarned > (degenBox.toAmount(MIM, degenBox.balanceOf(MIM, address(degenBoxCauldrons[i])), false))) {
                 MIM.transferFrom(mimProvider, address(degenBox), feesEarned);
                 degenBox.deposit(MIM, address(degenBox), address(degenBoxCauldrons[i]), feesEarned, 0);
-                mimFromDegenBox += feesEarned;
             }
             degenBoxCauldrons[i].withdrawFees();
         }
 
+        uint256 mimFromBentoBox = bentoBox.balanceOf(MIM, address(this));
+        uint256 mimFromDegenBox = degenBox.balanceOf(MIM, address(this));
         uint256 amountWithdrawn = mimFromBentoBox + mimFromDegenBox;
 
-        if (autoWithdrawFromBentoBoxes) {
-            withdrawFromBentoBoxes(mimFromBentoBox, mimFromDegenBox);
-
-            if (autoBridgeToEthereum) {
-                bridgeMimToEthereum(amountWithdrawn);
-            }
-        }
+        withdrawFromBentoBoxes(mimFromBentoBox, mimFromDegenBox);
+        bridgeMimToEthereum(amountWithdrawn);
```

Comment:
You need to do a conversion from share to amount or get the balanceOf from the mim contract for this address

Reply:
Therefore you also can’t sum the shares from DegenBox and BentoBox as the share to amount ratio will be different